File: tests/leetcode/x/c/0001.c
#include <stdio.h>
#include <stdlib.h>

static void two_sum(const int *nums, int n, int target, int *a, int *b) {
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (nums[i] + nums[j] == target) {
                *a = i;
                *b = j;
                return;
            }
        }
    }
    *a = 0;
    *b = 0;
}

int main(void) {
    int t;
    if (scanf("%d", &t) != 1) {
        return 0;
    }
    for (int tc = 0; tc < t; tc++) {
        int n, target;
        scanf("%d %d", &n, &target);
        int *nums = (int *)malloc((size_t)n * sizeof(int));
        for (int i = 0; i < n; i++) {
            scanf("%d", &nums[i]);
        }
        int a, b;
        two_sum(nums, n, target, &a, &b);
        printf("%d %d\n", a, b);
        free(nums);
    }
    return 0;
}

File: tests/leetcode/x/ocaml/0001.ml
let read_all () =
  let buf = Buffer.create 1024 in
  (try
     while true do
       Buffer.add_string buf (input_line stdin);
       Buffer.add_char buf ' '
     done
   with End_of_file -> ());
  Buffer.contents buf

let parse_ints s =
  let normalized =
    String.map
      (fun c -> if c = '\n' || c = '\r' || c = '\t' then ' ' else c)
      s
  in
  normalized
  |> String.split_on_char ' '
  |> List.filter (fun x -> x <> "")
  |> List.map int_of_string
  |> Array.of_list

let two_sum nums target =
  let n = Array.length nums in
  let rec outer i =
    if i >= n then (0, 0)
    else
      let rec inner j =
        if j >= n then outer (i + 1)
        else if nums.(i) + nums.(j) = target then (i, j)
        else inner (j + 1)
      in
      inner (i + 1)
  in
  outer 0

let () =
  let values = parse_ints (read_all ()) in
  if Array.length values > 0 then (
    let t = values.(0) in
    let idx = ref 1 in
    for tc = 0 to t - 1 do
      let n = values.(!idx) in
      incr idx;
      let target = values.(!idx) in
      incr idx;
      let nums = Array.init n (fun i -> values.(!idx + i)) in
      idx := !idx + n;
      let a, b = two_sum nums target in
      Printf.printf "%d %d" a b;
      if tc + 1 < t then print_newline ()
    done)
